aderyn_core/src/audit/send_ether_no_checks.rs

```diff
@@ -0,0 +1,132 @@
+use prettytable::{row, Row};
+
+use super::{auditor::AuditorDetector, investigators::SimpleInvestigator};
+use crate::{
+    context::workspace_context::WorkspaceContext, detect::helpers,
+    visitor::ast_visitor::ASTConstVisitor,
+};
+use std::error::Error;
+
+#[derive(Debug)]
+struct SendEtherNoChecksInstance {
+    pub filename: String,
+    pub line_no: usize,
+    pub func_name: String,
+}
+
+impl SendEtherNoChecksInstance {
+    fn encode_from(func_name: String, node_key: (String, usize, String)) -> Self {
+        let (filename, line_no, _) = node_key;
+        Self {
+            filename,
+            line_no,
+            func_name,
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct SendEtherWithoutMsgSenderChecksDetector {
+    found_instances: Vec<SendEtherNoChecksInstance>,
+}
+
+impl AuditorDetector for SendEtherWithoutMsgSenderChecksDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        for func in helpers::get_implemented_external_and_public_functions(context) {
+            let mut tracker = MsgSenderAndCallWithValueTracker::default();
+            let investigator = SimpleInvestigator::for_node(func, context)?;
```

Comment:
For my context: Does this create the CFG only on request from the Detector?

Is there a way we can do it (efficiently) globally as a preprocessing step so that all detectors can access it by default? How would that affect performance?

Reply:
💯 Global preprocessing is the goal but I thought Auditor-mode is a way to introduce it to aderyn, so that we can play with it for a while and see what can be done :)

Reply:
Yes, it creates the Callgraph on request from the detector

Reply:
✅ It's now a detector